docs/reference/manifest.md

```diff
@@ -0,0 +1,158 @@
+(chisel_manifest_ref)=
+# Chisel Manifest
+
+The Chisel manifest is a ZSTD-compressed file which lists the metadata about
+installed packages, slices and files. The uncompressed file is in the
+["jsonwall" format](chisel_manifest_format) - a collection of JSON objects,
+one-per-line.
+
+When building fine-grained and yet functional root file systems, Chisel creates
+this manifest as a way for ensuring file-level integrity and trace all the
+installed package slices, in a format that 3rd party tools (such as SBOM
+generators and vulnerability scanners) can work with.
+
+
+(chisel_manifest_location)=
+## Location of the manifest
+
+Chisel manifests may be generated anywhere in the newly created root file system.
+To specify the location, Chisel must be instructed to install a slice where at least
+one of its contents points to a path with the property
+{ref}`generate: manifest<slice_definitions_format_slices_contents_generate>`.
+
+When such a slice is installed, a `manifest.wall` file is generated at the specified
+path. If there are multiple paths of this kind being installed, a manifest will be
+created in each one of them.
+
+(chisel_manifest_pre-defined_location)=
+### Pre-defined location
+
+There is a pre-defined slice named `base-files_chisel` that is available in all
+supported Ubuntu releases in the {{chisel_releases_repo}}.
+
+
+```yaml
+package: base-files
+slices:
+  chisel:
+    contents:
+      /var/lib/chisel/**: {generate: manifest}
+  ...
+```
+
+Installing the `base-files_chisel` slice produces a manifest at
+`/var/lib/chisel/manifest.wall`.
+
+
+(chisel_manifest_format)=
+## Manifest format
+
+The uncompressed manifest is a "jsonwall" file. This is a custom database file
+format where there is one JSON object per line, and both the individual JSON
+fields and the lines themselves (except the
```

Comment:
This is not true, the lines are sorted according to the ordering of the fields in the struct and we define it manually so the fields that we want to use as indexes appear first.

Reply:
Indeed. I've rephrased it. Pls double check 75406e1

Reply:
I think I was not super clear in my previous comment. What I meant is that for one line, the contents of the struct, e.g. `{kind: path, mode: ...`} are ordered based on the ordering of the struct in the code, we define that based on the fields that we want to index. However, the file as a whole is ordered lexicographically so you can do binary search. That is, once you have all the lines, each one of them having the fields ordering defined in their respective struct, we sort all the lines to then output it as the jsonwall file (the lines are unchanged but their position in the file is sorted).

Reply:
But isn't that what the current text currently says? I.e. I'm not being specific about the struct-based JSON ordering, but I mentioned that the whole list is sorted (according to `kind` - which, because of the struct-based ordering, is always the 1st field)